src/engine/source/yml/src/yml.cpp

```diff
@@ -78,7 +73,18 @@ YAML::Node parse_scalar(const rapidjson::Value& node)
     return n;
 }
 
-YAML::Node json2yaml(const rapidjson::Value& value)
+rapidjson::Document Converter::loadYMLfromString(const std::string& yamlStr)
+{
+    YAML::Node root = YAML::Load(yamlStr);
+    rapidjson::Document doc;
+
+    rapidjson::Value val = yaml2json(root, doc.GetAllocator());
```

Comment:
could parsing fail?

Reply:
The requested analysis is beyond the scope of this pull request, please report it to the appropriate party